Tasks/Src/Task_CheckContactor.c

```diff
@@ -1,16 +1,56 @@
 /* Copyright (c) 2018-2022 UT Longhorn Racing Solar */
-#include "RTOS_BPS.h"
+#include "config.h"
 #include "Contactor.h"
+#include "CANbus.h"
 #include "Tasks.h"
+#include "BSP_UART.h"
+#include "Voltage.h"
+#include "CAN_Queue.h"
 
-/* 
- * Launch this task once the contactor has been closed.
- * This will trip the BPS if the contactor is open.
- * The purpose of this is to trip the BPS if the emergency
- * cutoff switch is pressed.
- */
 void Task_CheckContactor(void *p_arg) {
     (void)p_arg;
+    
+    CANMSG_t CANMSG;
+    CANPayload_t CanPayload; 
+    CanPayload.idx    = 0;
+    CanPayload.data.b = 1;
+
+    // If a contactor is on before we turn it on in this task, it may have failed and welded shut
+    if(Contactor_GetState(HVHIGH_CONTACTOR) || Contactor_GetState(HVLOW_CONTACTOR)) {
+        Fault_BitMap |= Fault_ESTOP;
+        EnterFaultState();
+    } else {
+        RTOS_BPS_SemPost(&SafetyCheck_Sem4, OS_OPT_POST_1);
+    }
+
+    // BLOCKING =====================
+    // Wait until voltage, open wire, temperature, contactors, and current(Amperes) are all checked and safe
+    for (int i = 0; i < NUM_FAULT_POINTS; i++){
+        RTOS_BPS_SemPend(&SafetyCheck_Sem4,OS_OPT_PEND_BLOCKING);
```

Comment:
these pends happen before the watchdog is launched, so a hang on initial safety checks won't be caught -- it'll be helpful to add a timeout on the ```SafetyCheck_Sem4``` checks and to fault on timeout.

Reply:
We should update the function to pass in a timeout. Or you can create another function that is just for semaphore pending with timeout. Both would work, its just preference on coding style. In the first case, you would have to pass in NULL in 90% of cases where the function is used and update all the function calls. In the second case, you would have an almost duplicate function.